signer/tests/integration/contracts.rs

```diff
@@ -79,63 +101,123 @@ where
+                keys: key_pairs,
+                stacks_client: client,
+            }
+        })
+        .await;
 
     SBTC_DEPLOYMENT
         .get_or_init(|| async {
             // The registry and token contracts need to be deployed first
             // and second respectively. The rest can be deployed in any
             // order.
-            deploy_smart_contract(&state, &client, SbtcRegistryContract).await;
-            deploy_smart_contract(&state, &client, SbtcTokenContract).await;
-            deploy_smart_contract(&state, &client, SbtcDepositContract).await;
-            deploy_smart_contract(&state, &client, SbtcWithdrawalContract).await;
-            deploy_smart_contract(&state, &client, SbtcBootstrapContract).await;
-            true
+            deploy_smart_contract(signer, SbtcRegistryContract).await;
+            deploy_smart_contract(signer, SbtcTokenContract).await;
+            deploy_smart_contract(signer, SbtcDepositContract).await;
+            deploy_smart_contract(signer, SbtcWithdrawalContract).await;
+            deploy_smart_contract(signer, SbtcBootstrapContract).await;
         })
         .await;
 
-    state
+    signer
 }
 
 #[ignore]
+#[test_case(ContractCall(CompleteDepositV1 {
+    outpoint: bitcoin::OutPoint::null(),
+    amount: 123654,
+    recipient: StacksAddress::from_string("ST1RQHF4VE5CZ6EK3MZPZVQBA0JVSMM9H5PMHMS1Y").unwrap(),
+    deployer: testing::wallet::generate_wallet().0.address(),
+}); "complete-deposit")]
+#[test_case(ContractCall(AcceptWithdrawalV1 {
+    request_id: 0,
+    outpoint: bitcoin::OutPoint::null(),
+    tx_fee: 3500,
+    signer_bitmap: BitArray::new([0; 2]),
+    deployer: testing::wallet::generate_wallet().0.address(),
+}); "accept-withdrawal")]
+#[test_case(ContractCall(RejectWithdrawalV1 {
+    request_id: 0,
+    signer_bitmap: BitArray::new([0; 2]),
+    deployer: testing::wallet::generate_wallet().0.address(),
+}); "reject-withdrawal")]
+#[test_case(ContractCall(RotateKeysV1::new(
+    &testing::wallet::generate_wallet().0,
+    testing::wallet::generate_wallet().0.address(),
+)); "rotate-keys")]
 #[tokio::test]
-async fn test_deploy() {
-    let _state = deploy_smart_contracts().await;
-}
+async fn complete_deposit_wrapper_tx_accepted<T: AsContractCall>(contract: ContractCall<T>) {
+    let signer = deploy_smart_contracts().await;
+    let mut unsigned = MultisigTx::new_tx(contract, &signer.wallet, TX_FEE);
 
-#[ignore]
-#[tokio::test]
-async fn complete_deposit_wrapper_tx_accepted() {
-    // TODO(#264): Add integration test for signing Stacks smart contracts
-}
+    for signature in make_signatures(unsigned.tx(), &signer.keys) {
+        unsigned.add_signature(signature).unwrap();
+    }
+    let tx = unsigned.finalize_transaction();
 
-#[ignore]
-#[tokio::test]
-async fn accept_withdrawal_request_tx_accepted() {
-    // TODO(#264): Add integration test for signing Stacks smart contracts
-}
+    tokio::time::sleep(std::time::Duration::from_secs(5)).await;
```

Comment:
What are we waiting for here? Could you add a comment to describe. I assume this is for the smart contract deployments to be mined right?

Reply:
Yeah, it's to wait for the smart contract to be mined. I'll add a comment.

Reply:
Right, thank you!